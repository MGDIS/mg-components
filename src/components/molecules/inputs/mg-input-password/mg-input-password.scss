@use '../styles/mg-input.scss';
@use '../styles/mg-input-box.scss';
@use '../styles/mg-input-width.scss';
@use '../../../../styles/a11y.scss';
