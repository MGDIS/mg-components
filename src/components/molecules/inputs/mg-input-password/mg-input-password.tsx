import { Component, Event, h, Prop, EventEmitter, State, Watch } from '@stencil/core';
import { MgInput } from '../MgInput';
import { createID, ClassList } from '../../../../utils/components.utils';
import { messages } from '../../../../locales';

@Component({
  tag: 'mg-input-password',
  styleUrl: 'mg-input-password.scss',
  shadow: true,
})
export class MgInputPassword {
  /************
   * Internal *
   ************/

  // classes
  private classError = 'is-not-valid';

  // HTML selector
  private input: HTMLInputElement;

  /**************
   * Decorators *
   **************/

  /**
   * Component value
   */
  @Prop({ mutable: true, reflect: true }) value: string;

  /**
   * Identifier is used for the element ID (id is a reserved prop in Stencil.js)
   * If not set, it will be created.
   */
  @Prop() identifier?: string = createID('mg-input-password');

  /**
   * Input name
   * If not set the value equals the identifier
   */
  @Prop() name?: string = this.identifier;

  /**
   * Input label
   * Required
   */
  @Prop() label!: string;

  /**
   * Define if label is displayed on top
   */
  @Prop() labelOnTop: boolean;

  /**
   * Define if label is visible
   */
  @Prop() labelHide: boolean = false;

  /**
   * Input placeholder.
   * It should be a word or short phrase that demonstrates the expected type of data, not a replacement for labels or help text.
   */
  @Prop() placeholder: string;

  /**
   * Define if input is required
   */
  @Prop() required: boolean = false;

  /**
   * Define if input is readonly
   */
  @Prop() readonly: boolean = false;
  @Watch('readonly')
  handleReadOnly() {
    this.classList.delete(this.classError);
  }
  /**
   * Define if input is disabled
   */
  @Prop() disabled: boolean = false;
  @Watch('disabled')
  handleDisabled() {
    this.classList.delete(this.classError);
  }
  /**
   * Add a tooltip message next to the input
   */
  @Prop() tooltip: string;

  /**
   * Add a help text under the input, usually expected data format and example
   */
  @Prop() helpText: string;

  /**
   * Define input pattern to validate
   */
  @Prop({ mutable: true }) valid: boolean;

  /**
   * Define input pattern error message
   */
  @Prop({ mutable: true }) invalid: boolean;

  /**
   * Component classes
   */
  @State() classList: ClassList = new ClassList(['mg-input--password', 'mg-input--width-full']);

  /**
   * Error message to display
   */
  @State() errorMessage: string;

  /**
   * Emmited event when value change
   */
  @Event({ eventName: 'value-change' }) valueChange: EventEmitter<string>;

  /**
   * Handle input event
   */
  private handleInput = () => {
    this.checkValidity();
    this.value = this.input.value;
    this.valueChange.emit(this.value);
  };

  /**
   * Handle blur event
   */
  private handleBlur = () => {
    this.checkValidity();
    this.checkError();
  };

  /**
   * Check if input is valid
   */
  private checkValidity() {
    if (!this.readonly && this.input !== undefined) {
      const validity = this.input.checkValidity();

      // Set validity
      this.valid = validity;
      this.invalid = !validity;
    }
  }

  /**
   * Check input errors
   */
  private checkError() {
    // Set error message
    this.errorMessage = undefined;
    // required
    if (!this.valid && this.input.validity.valueMissing) {
      this.errorMessage = messages.errors.required;
    }
    // Update class
    if (this.valid) {
      this.classList.delete(this.classError);
    } else {
      this.classList.add(this.classError);
    }
  }

  /*************
   * Lifecycle *
   *************/

  componentWillLoad() {
    // return a promise to process action only in the FIRST render().
    // https://stenciljs.com/docs/component-lifecycle#componentwillload
    return setTimeout(() => this.checkValidity(), 0);
  }

  render() {
    return (
      <MgInput
        identifier={this.identifier}
        classList={this.classList}
        label={this.label}
        labelOnTop={this.labelOnTop}
        labelHide={this.labelHide}
        required={this.required}
        readonly={this.readonly}
        value={this.value}
        readonlyValue={this.value !== undefined ? 'â€¢'.repeat(this.value.length) : undefined}
        tooltip={this.tooltip}
        displayCharacterLeft={false}
        characterLeftTemplate={undefined}
        maxlength={undefined}
        helpText={this.helpText}
        errorMessage={this.errorMessage}
        isFieldset={false}
      >
        <input
          type="password"
          class="mg-input__box"
          value={this.value}
          id={this.identifier}
          name={this.name}
          placeholder={this.placeholder}
          title={this.placeholder}
          disabled={this.disabled}
          required={this.required}
          onInput={this.handleInput}
          onBlur={this.handleBlur}
          ref={el => (this.input = el as HTMLInputElement)}
        />
      </MgInput>
    );
  }
}
