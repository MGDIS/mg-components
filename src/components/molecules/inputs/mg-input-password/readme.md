This molecule refers to the [PDA9-769](https://jira.mgdis.fr/browse/PDA9-769)

<!-- Auto Generated Below -->


## Properties

| Property             | Attribute      | Description                                                                                                                                    | Type                     | Default                         |
| -------------------- | -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------ | ------------------------------- |
| `disabled`           | `disabled`     | Define if input is disabled                                                                                                                    | `boolean`                | `false`                         |
| `helpText`           | `help-text`    | Add a help text under the input, usually expected data format and example                                                                      | `string`                 | `undefined`                     |
| `identifier`         | `identifier`   | Identifier is used for the element ID (id is a reserved prop in Stencil.js) If not set, it will be created.                                    | `string`                 | `createID('mg-input-password')` |
| `invalid`            | `invalid`      | Define input invalid state                                                                                                                     | `boolean`                | `undefined`                     |
| `label` _(required)_ | `label`        | Input label Required                                                                                                                           | `string`                 | `undefined`                     |
| `labelHide`          | `label-hide`   | Define if label is visible                                                                                                                     | `boolean`                | `false`                         |
| `labelOnTop`         | `label-on-top` | Define if label is displayed on top                                                                                                            | `boolean`                | `undefined`                     |
| `name`               | `name`         | Input name If not set the value equals the identifier                                                                                          | `string`                 | `this.identifier`               |
| `placeholder`        | `placeholder`  | Input placeholder. It should be a word or short phrase that demonstrates the expected type of data, not a replacement for labels or help text. | `string`                 | `undefined`                     |
| `readonly`           | `readonly`     | Define if input is readonly                                                                                                                    | `boolean`                | `false`                         |
| `required`           | `required`     | Define if input is required                                                                                                                    | `boolean`                | `false`                         |
| `tooltip`            | `tooltip`      | Add a tooltip message next to the input                                                                                                        | `string`                 | `undefined`                     |
| `valid`              | `valid`        | Define input valid state                                                                                                                       | `boolean`                | `undefined`                     |
| `value`              | `value`        | Component value                                                                                                                                | `string`                 | `undefined`                     |
| `width`              | `width`        | Define input width                                                                                                                             | `"full" \| 16 \| 2 \| 4` | `'full'`                        |


## Events

| Event          | Description                         | Type                   |
| -------------- | ----------------------------------- | ---------------------- |
| `input-valid`  | Emited event when checking validity | `CustomEvent<boolean>` |
| `value-change` | Emited event when value change      | `CustomEvent<string>`  |


## Methods

### `displayError() => Promise<void>`

Public method to display errors

#### Returns

Type: `Promise<void>`




## Dependencies

### Depends on

- [mg-tooltip](../../../atoms/mg-tooltip)
- [mg-icon](../../../atoms/mg-icon)
- [mg-input-title](../../../atoms/mg-input-title)
- [mg-character-left](../../../atoms/mg-character-left)

### Graph
```mermaid
graph TD;
  mg-input-password --> mg-tooltip
  mg-input-password --> mg-icon
  mg-input-password --> mg-input-title
  mg-input-password --> mg-character-left
  style mg-input-password fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
