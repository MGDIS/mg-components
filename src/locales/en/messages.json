{
  "errors": {
    "required": "This field is required.",
    "date": {
      "badInput": "The date format must be <span aria-hidden=\"true\">mm/dd/yyyy</span><span class=\"sr-only\">m m / d d / y y y y</span> and the date must be greater than {min}",
      "min": "The date cannot be earlier than {min}",
      "max": "The date cannot be later than {max}",
      "minMax": "The date must be between {min} and {max}"
    },
    "numeric": {
      "min": "The value cannot be less than {min}",
      "max": "The value cannot be greater than {max}",
      "minMax": "The value must be between {min} et {max}"
    }
  },
  "nbCharLeft": "{counter} characters left.",
  "input": {
    "select": {
      "placeholder": "Select a value"
    }
  },
  "form": {
    "required": "Fields with a <strong class=\"is-asterisk\">*</strong> are required",
    "allRequired": "All fields are required"
  },
  "message": {
    "closeButton": "Close message"
  },
  "modal": {
    "closeButton": "Close modal"
  },
  "pagination": {
    "label": "Pagination",
    "page": "page",
    "pages": "pages",
    "nextPage": "Next page",
    "previousPage": "Previous page",
    "selectPage": "Select the page to display."
  },
  "panel": {
    "editLabel": "Edit section title."
  },
  "general": {
    "confirm": "Confirm",
    "cancel": "Cancel",
    "close": "Close",
    "next": "next",
    "previous": "previous"
  }
}
