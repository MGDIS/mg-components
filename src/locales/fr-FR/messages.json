{
  "errors": {
    "required": "Ce champ est obligatoire.",
    "date": {
      "badInput": "Le format de la date doit être du type <span aria-hidden=\"true\">jj/mm/aaaa</span><span class=\"sr-only\">j j / m m / a a a a</span> et la date supérieure au {min}",
      "min": "La date ne peut pas être antérieure au {min}",
      "max": "La date ne peut pas être postérieure au {max}",
      "minMax": "La date doit être comprise entre {min} et {max}"
    },
    "numeric": {
      "min": "La valeur ne peut pas être inférieure à {min}",
      "max": "La valeur ne peut pas être supérieure à {max}",
      "minMax": "La valeur doit être comprise entre {min} et {max}"
    }
  },
  "nbCharLeft": "{counter} caractères disponibles.",
  "input": {
    "select": {
      "placeholder": "Sélectionnez une valeur"
    },
    "checkbox": {
      "true": "Oui",
      "false": "Non",
      "indeterminate": "Indéterminé"
    }
  },
  "form": {
    "required": "Les champs marqués d'un <strong class=\"is-asterisk\">*</strong> sont obligatoires",
    "allRequired": "Tous les champs sont obligatoires"
  },
  "message": {
    "closeButton": "Fermer le message"
  },
  "popover": {
    "closeButton": "Fermer"
  },
  "pagination": {
    "page": "page",
    "pages": "pages",
    "next": "suivant",
    "previous": "précédent",
    "nextPage": "Page suivante",
    "previousPage": "Page précédente",
    "selectPage": "Sélectionner la page à afficher."
  }
}
