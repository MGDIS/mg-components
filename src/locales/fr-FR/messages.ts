const messages = {
  "errors": {
    "required": "Ce champs est obligatoire.",
    "date": {
      "badInput": "Le format de la date doit être du type <span aria-hidden=\"true\">jj/mm/aaaa</span><span class=\"sr-only\">j j / m m / a a a a</span> et supérieure au 01/01/1900",
    },
    "currency": {
      "min": "La valeur ne peut pas être inférieure à {min}",
      "max": "La valeur ne peut pas être supérieure à {max}",
      "minMax": "La valeur doit être comprise entre {min} et {max}",
    }
  },
  "nbCharLeft" : "{counter} caractères disponibles.",
  "input": {
    "select" : {
      "placeholder": "Veuillez sélectionner une valeur"
    },
    "checkbox": {
      "true": "Oui",
      "false": "Non",
      "indeterminate": "Indéterminé",
    },
  },
  "message": {
    "closeButton": "Fermer le message",
  }
}

export default messages;
