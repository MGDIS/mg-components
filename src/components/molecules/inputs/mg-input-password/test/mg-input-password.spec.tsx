import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgInputPassword } from '../mg-input-password';
import messages from '../../../../../locales/en/messages.json';

const getPage = args => {
  const page = newSpecPage({
    components: [MgInputPassword],
    template: () => <mg-input-password {...args}></mg-input-password>,
  });
  jest.runAllTimers();
  return page;
};

describe('mg-input-password', () => {
  beforeEach(() => jest.useFakeTimers());
  afterEach(() => jest.runOnlyPendingTimers());
  test.each([
    { label: 'label', identifier: 'identifier' },
    { label: 'label', identifier: 'identifier', labelHide: true },
    { label: 'label', identifier: 'identifier', labelOnTop: true },
    { label: 'label', identifier: 'identifier', readonly: true },
    { label: 'label', identifier: 'identifier', readonly: true, labelOnTop: true, tooltip: 'Tooltip message' },
    { label: 'label', identifier: 'identifier', readonly: true, value: 'blu' },
    { label: 'label', identifier: 'identifier', tooltip: 'My Tooltip Message' },
    { label: 'label', identifier: 'identifier', tooltip: 'My Tooltip Message', labelOnTop: true },
  ])('Should render with args %s:', async args => {
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });

  test.each(['', undefined])('Should throw error with invalid label property : %s', async value => {
    try {
      await getPage({ label: value });
    } catch (err) {
      expect(err.message).toMatch('<mg-input> prop "label" is required');
    }
  });

  test('Should throw an error with labelOnTop & labelHide set to true', async () => {
    try {
      await getPage({ label: 'batman', labelOnTop: true, labelHide: true });
    } catch (err) {
      expect(err.message).toMatch('<mg-input> prop "labelOnTop" must not be paired with the prop "labelHide"');
    }
  });

  test('Should trigger events', async () => {
    const inputValue = 'Blu';
    const args = { label: 'label', identifier: 'identifier', helpText: 'My help text' };
    const page = await getPage(args);

    const element = page.doc.querySelector('mg-input-password');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(() => true);
    Object.defineProperty(input, 'validity', {
      get: jest.fn(() => ({
        valueMissing: false,
        patternMismatch: false,
      })),
    });

    jest.spyOn(page.rootInstance.valueChange, 'emit');

    input.dispatchEvent(new CustomEvent('focus', { bubbles: true }));
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot(); //Snapshot on focus

    input.value = inputValue;
    input.dispatchEvent(new CustomEvent('input', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.valueChange.emit).toHaveBeenCalledWith(inputValue);
  });

  describe.each(['readonly', 'disabled'])('validity, case next state is %s', nextState => {
    test.each([
      { validity: true, valueMissing: false },
      { validity: false, valueMissing: true },
      { validity: false, valueMissing: false },
    ])('validity (%s), valueMissing (%s)', async ({ validity, valueMissing }) => {
      const args = { label: 'label', identifier: 'identifier', patternErrorMessage: 'Non' };
      const page = await getPage(args);

      const element = page.doc.querySelector('mg-input-password');
      const input = element.shadowRoot.querySelector('input');

      //mock validity
      input.checkValidity = jest.fn(() => validity);
      Object.defineProperty(input, 'validity', {
        get: jest.fn(() => ({
          valueMissing,
        })),
      });

      input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
      await page.waitForChanges();

      if (validity) {
        expect(page.rootInstance.errorMessage).toBeUndefined();
      } else if (valueMissing) {
        expect(page.rootInstance.errorMessage).toEqual(messages.errors.required);
      }
      expect(page.rootInstance.valid).toEqual(validity);
      expect(page.rootInstance.invalid).toEqual(!validity);
      if (valueMissing) {
        expect(page.root).toMatchSnapshot(); //Snapshot with readonly/disabled FALSE
        element[nextState] = true;
        await page.waitForChanges();
        expect(page.root).toMatchSnapshot(); //Snapshot with readonly/disabled TRUE
      }
    });
  });

  test("display error with displayError component's public method", async () => {
    const page = await getPage({ label: 'label', identifier: 'identifier', required: true });

    expect(page.root).toMatchSnapshot();

    const element = page.doc.querySelector('mg-input-password');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(() => false);
    Object.defineProperty(input, 'validity', {
      get: jest.fn(() => ({
        valueMissing: true,
      })),
    });

    await element.displayError();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  test.each(['fr', 'xx'])('display error message with locale: %s', async lang => {
    const page = await getPage({ label: 'label', identifier: 'identifier', required: true, lang });
    const element = page.doc.querySelector('mg-input-password');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(() => false);
    Object.defineProperty(input, 'validity', {
      get: jest.fn(() => ({
        valueMissing: true,
      })),
    });

    await element.displayError();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  test('Should remove error on input', async () => {
    const page = await getPage({ label: 'label', identifier: 'identifier', required: true });
    const element = page.doc.querySelector('mg-input-password');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn().mockReturnValueOnce(false).mockReturnValueOnce(true).mockReturnValueOnce(true);
    Object.defineProperty(input, 'validity', {
      get: jest
        .fn()
        .mockReturnValueOnce({
          valueMissing: true,
        })
        .mockReturnValueOnce({
          valueMissing: true,
        })
        .mockReturnValueOnce({
          valueMissing: false,
        })
        .mockReturnValueOnce({
          valueMissing: false,
        }),
    });

    await element.displayError();

    await page.waitForChanges();

    expect(page.rootInstance.hasError).toBeTruthy();
    expect(page.rootInstance.errorMessage).toEqual(messages.errors.required);

    input.value = 'blu';
    input.dispatchEvent(new CustomEvent('input', { bubbles: true }));
    await page.waitForChanges();

    expect(page.rootInstance.hasError).toBeTruthy();
    expect(page.rootInstance.errorMessage).toBeUndefined();

    input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
    await page.waitForChanges();

    expect(page.rootInstance.hasError).toBeFalsy();
  });
});
