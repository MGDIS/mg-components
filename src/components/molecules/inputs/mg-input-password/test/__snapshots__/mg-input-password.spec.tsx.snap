// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-input-password Should render with args {
  label: 'label',
  identifier: 'identifier',
  readonly: true,
  labelOnTop: true,
  tooltip: 'Tooltip message'
}: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--label-on-top mg-input--password mg-input--readonly mg-input--width-full">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
      </div>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args {
  label: 'label',
  identifier: 'identifier',
  readonly: true,
  value: 'blu'
}: 1`] = `
<mg-input-password value="blu">
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--readonly mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong>
          •••
        </strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args {
  label: 'label',
  identifier: 'identifier',
  tooltip: 'My Tooltip Message'
}: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" type="password">
          <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
            <mg-icon icon="info-circle"></mg-icon>
          </mg-tooltip>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args {
  label: 'label',
  identifier: 'identifier',
  tooltip: 'My Tooltip Message',
  labelOnTop: true
}: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--label-on-top mg-input--password mg-input--width-full">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
        <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
          <mg-icon icon="info-circle"></mg-icon>
        </mg-tooltip>
      </div>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args { label: 'label', identifier: 'identifier' }: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args { label: 'label', identifier: 'identifier', labelHide: true }: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--width-full">
      <mg-input-title class="sr-only" identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args { label: 'label', identifier: 'identifier', labelOnTop: true }: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--label-on-top mg-input--password mg-input--width-full">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
      </div>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should render with args { label: 'label', identifier: 'identifier', readonly: true }: 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--readonly mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password Should trigger events 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-help-text" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
        <div class="mg-input__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password display error message with locale: fr 1`] = `
<mg-input-password lang="fr">
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" required="" type="password">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          Ce champ est obligatoire.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password display error message with locale: xx 1`] = `
<mg-input-password lang="xx">
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" required="" type="password">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password display error with displayError component's public method 1`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" required="" type="password">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password display error with displayError component's public method 2`] = `
<mg-input-password>
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" required="" type="password">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password validity, case next state is disabled validity ({ validity: false, valueMissing: true }), valueMissing (%s) 1`] = `
<mg-input-password patternerrormessage="Non">
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password validity, case next state is disabled validity ({ validity: false, valueMissing: true }), valueMissing (%s) 2`] = `
<mg-input-password patternerrormessage="Non">
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" disabled="" id="identifier" name="identifier" type="password">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password validity, case next state is readonly validity ({ validity: false, valueMissing: true }), valueMissing (%s) 1`] = `
<mg-input-password patternerrormessage="Non">
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--password mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" type="password">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;

exports[`mg-input-password validity, case next state is readonly validity ({ validity: false, valueMissing: true }), valueMissing (%s) 2`] = `
<mg-input-password patternerrormessage="Non">
  <mock:shadow-root>
    <div class="mg-input mg-input--password mg-input--readonly mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-password>
`;
